quickstarts/microsoft.digitaltwins/digitaltwins-with-function-private-link/main.bicep

```diff
@@ -0,0 +1,93 @@
+@description('The location into which the Azure Storage resources should be deployed.')
+@allowed([
+  'westcentralus'
+  'westus2'
+  'northeurope'
+  'australiaeast'
+  'westeurope'
+  'eastus'
+  'southcentralus'
+  'southeastasia'
+  'uksouth'
+  'eastus2'
+])
+param location string
+
+@description('Azure Digital Twins instance name')
+param digitalTwinsInstanceName string = 'digitaltwins-${uniqueString(resourceGroup().name)}'
+
+@description('Azure function name')
+@maxLength(16)
+param functionName string ='${uniqueString(resourceGroup().name)}'
+
+@description('Virtual Network name')
+param virtualNetworkName string = 'vnet-${uniqueString(resourceGroup().name)}'
+
+@description('Virtual Network Address Prefix')
+param vnetAddressPrefix string = '10.0.0.0/22'
+
+@description('Function Subnet Address Prefix')
+param functionAddressPrefix string = '10.0.0.0/24'
+
+@description('Private Link Subnet Address Prefix')
+param privateLinkAddressPrefix string = '10.0.1.0/24'
+
+var privateLinkSubnetName = 'PrivateLinkSubnet'
+var functionSubnetName = 'FunctionSubnet'
+
+module digitaltwins 'modules/digitaltwins.bicep' = {
+  name: 'digitaltwins'
+  params: {
+    digitalTwinsInstanceName: digitalTwinsInstanceName
+    digitalTwinsInstanceLocation: location
+  }
+}
+
+module network 'modules/network.bicep' = {
+  name: 'network'
+  params: {
+    virtualNetworkName: virtualNetworkName
+    virtualNetworkLocation: location
+    virtualNetworkAddressPrefix: vnetAddressPrefix
+    functionSubnetName: functionSubnetName
+    functionSubnetPrefix: functionAddressPrefix
+    privateLinkSubnetName: privateLinkSubnetName
+    privateLinkSubnetPrefix: privateLinkAddressPrefix
+  }
+}
+
+module privatelink 'modules/privatelink.bicep' = {
+  name: 'privatelink'
+  params: {
+    privateLinkName: 'PrivateLinkToDigitalTwins'
+    location: location
+    privateLinkServiceResourceId: digitaltwins.outputs.id
+    groupId: 'API'
+    privateLinkSubnetName: privateLinkSubnetName
+    privateDnsZoneName: 'privatelink.digitaltwins.azure.net'
+    virtualNetworkResourceName: virtualNetworkName
+  }
+}
+
+module function 'modules/function.bicep' = {
+  name: 'function'
+  params: {
+    location: location
+    virtualNetworkName: virtualNetworkName
+    storageAccoutName: '${toLower(functionName)}stg'

```

Comment:
function names can contain chars that are not allowed in storageAccounts - need to filter those or just use a name.